stdlib/public/runtime/Metadata.cpp

```diff
@@ -56,19 +62,45 @@
 using namespace swift;
 using namespace metadataimpl;
 
+// allocate memory up to a nearby page boundary
+static void *swift_allocPage(size_t size) {
```

Comment:
The current implementation does assume that `swift_allocPage` returns page-aligned memory.